core/src/main/scala/org/typelevel/sbt/BspControlPlugin.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.sbt
+
+import sbt.Keys._
+import sbt._
+import sbtcrossproject.CrossPlugin
+import sbtcrossproject.CrossPlugin.autoImport._
+import sbtcrossproject.Platform
+
+object BspControlPlugin extends AutoPlugin {
+  override def trigger = allRequirements
+  override def requires: Plugins = CrossPlugin
+
+  object autoImport {
+    lazy val tlBspPlatforms: SettingKey[Set[Platform]] =
+      settingKey[Set[Platform]](
+        "set of platforms for which BSP is enabled (default: JVMPlatform only)")
+  }
+
+  import autoImport._
+
+  override def globalSettings: Seq[Setting[?]] = Seq(
+    tlBspPlatforms := Set(JVMPlatform)
```

Comment:
I wish we didn't have to favor the `JVMPlatform` so blatantly here. It makes sense for many builds, but can also be less-than-ideal for some builds.

1. Projects defined as e.g. `crossProject(JSPlatform)`. Yeah this is dumb, but I see it all the time.
2. Cross-projects that have significant non-JVM sources. E.g. CE core, fs2-io.
3. The mythical `crossProject(JSPlatform, NativePlatform)`. I don't think I've ever seen this in the wild.

Reply:
changed to `tlBspExcludePlatforms` with empty set by default.